src/wasm-binary.h

```diff
@@ -502,57 +534,81 @@ class WasmBinaryWriter : public WasmVisitor<void> {
   }
 
   void emitString(const char *str) {
+    if (debug) std::cerr << "emitString " << str << std::endl;
     emitBuffer(str, strlen(str) + 1);
   }
 
   void finishUp() {
+    if (debug) std::cerr << "finishUp" << std::endl;
     // finish buffers
     for (auto& buffer : buffersToWrite) {
+      if (debug) std::cerr << "writing buffer" << (int)buffer.data[0] << "," << (int)buffer.data[1] << " at " << o.size() << " and pointer is at " << buffer.pointerLocation << std::endl;
       o.writeAt(buffer.pointerLocation, (uint32_t)o.size());
       for (size_t i = 0; i < buffer.size; i++) {
-        o << buffer.data[i];
+        o << (uint8_t)buffer.data[i];
       }
     }
   }
 
   // AST writing via visitors
 
+  int depth;
+
+  void recurse(Expression*& curr) {
+    if (debug) std::cerr << "zz recurse into " << ++depth << " at " << o.size() << std::endl;
+    visit(curr);
+    if (debug) std::cerr << "zz recurse from " << depth-- << " at " << o.size() << std::endl;
```

Comment:
Inc/dec of `depth` in `debug` code is correct? Doesn't that change what `visit(curr)` do depending on whether `debug` is on or not?